src/methods/SANA.cpp

```diff
@@ -1835,20 +1835,52 @@ uint SANA::getHighestIndex() const {
     return highestIndex;
 }
 
-#define LOG10_LOW_TEMP -10.0
-#define LOG10_HIGH_TEMP 10.0
-#define LOG10_NUM_STEPS 20 // looping through 0 <= i <= LOG10_NUM_STEPS
+#define HIGH_TEMP_LIMIT 0.99999
+#define LOW_TEMP_LIMIT 1e-10
+
+
+double SANA::temperatureBracket(double LIMIT){
+	int i = 0;
+	while (true){
+		double pBadi = pForTInitial(pow(10,i));
+		double pBadi1 = pForTInitial(pow(10,i+1));
+		cout << "Testing 10 ^ " << i << " pBad = " << pBadi << endl;
+		if (pBadi < LIMIT && pBadi1 >= LIMIT){
+			break;
+		}else if (pBadi > LIMIT){
+			i--;
+		}
+		else if(pBadi < LIMIT){
+			i++;
+		}
+	}
+	return LIMIT == HIGH_TEMP_LIMIT ? i+1 : i;
```

Comment:
It'll almost never be *equal*; choose a < or > here.

Reply:
Yes, I agree it won't be. But here the LIMIT just means the defined limits, either 0.9999 or 1e^-10. Following your psuedocode, if we are checking for High temperature limit, then we want to return i+1, else for Low temperature limit we want to return just i (since i is negative here, i+1 will change it in the wrong direction).